builtin/providers/openstack/resource_openstack_networking_router_v2.go

```diff
@@ -227,3 +233,11 @@ func waitForRouterDelete(networkingClient *gophercloud.ServiceClient, routerId s
 		return r, "ACTIVE", nil
 	}
 }
+
+func routerDriverOpts(d *schema.ResourceData) map[string]string {
+	m := make(map[string]string)
+	for key, val := range d.Get("driver_opts").(map[string]interface{}) {
```

Comment:
Does this throw an error if driver_opts is empty? I'm just being overly cautious from the bug I caused last week by doing an unsafe type cast.

I wasn't able to trigger the bug until I actually compiled Terraform and ran it with a new configuration. Acceptance tests and existing configurations failed to trigger the bug.